part-2.py

```diff
@@ -4,13 +4,45 @@
 # the appropriate comment.
 
 # search
-
+def search(array, query):
+    if not len(array):
+        return False
+    if array[0] == query:
+        return True
+    return search(array[1:], query)
 
 
 # is_palindrome
-
+def is_palindrome(text):
+    if len(text) == 1:
+        return True
+    if text[0] != text[-1]:
+        return False
+    return is_palindrome(text[1:-1])
 
 
 # digit_match
+# i felt pretty good about the other problems
+# but felt COMPLETELY lost with this one
+# i'm still not sure why or how i got it to work
+# and i'm sure i have unnecessary things here
```

Comment:
It looks good! One way to consider drying up your code is to add 0 to 1 to your recursive call with `bool(a_digit == b_digit)` which will convert the boolean to a 0 or a 1. You may also consider moving `if a < 10 or b < 10` outside of the section from lines 41-47 to make it clear it is a base case.

Reply:
Another option for this problem is to convert the input to a string.